pkg/controller/clusterdeployment/clusterdeployment_controller.go

```diff
@@ -859,6 +864,13 @@ func (r *ReconcileClusterDeployment) syncDeletedClusterDeployment(cd *hivev1.Clu
 			_, err := controllerutils.SetupClusterInstallServiceAccount(r, cd.Namespace, cdLog)
 			if err != nil {
 				cdLog.WithError(err).Error("error setting up service account and role")
+				if strings.Contains(err.Error(), terminatedErrorString) {
```

Comment:
@dgoodwin should we also confirm that the namespace is getting deleted by doing a get on the namespace, instead of just relying on the error string?